cob_monitoring/src/battery_monitor.py

```diff
@@ -152,53 +174,84 @@ def say(self, text):
             for component in self.sound_components:
                 sss.say(component, [text])
 
+    def trigger_horn(self):
+
+        if self.enable_horn != True:
+            raise ValueError(f"could not signal horn, not enabled")
+
+        req = SetHornModeRequest()
+        req.mode.mode = HornMode.FLASH
+        req.mode.frequency = self.horn_frequency
+        req.mode.pulses = self.horn_pulses
+        req.mode.timeout = self.horn_timeout
+        try:
+            self.srv_set_horn.call(req)
+        except (TypeError, rospy.ServiceException) as e:
+            rospy.logwarn(f"Received exception when calling '{self.set_horn_service}': {e}")
+            return
+
     def timer_callback(self, event):
         # warn if battery is empty
         if not self.is_charging:
-            # 5%
-            if self.power_state.relative_remaining_capacity <= self.threshold_critical and (rospy.get_time() - self.last_time_warned) > self.interval_duration_critical:
-                self.last_time_warned = rospy.get_time()
-                mode = copy.copy(self.mode)
-                mode.mode = LightModes.FLASH
-                color = self.color_critical
-                mode.colors = []
-                mode.colors.append(color)
-                mode.frequency = 4
-                mode.pulses = 8
-                self.set_light(mode)
 
-                self.say(self.sound_critical)
+            # 5%
+            if self.power_state.relative_remaining_capacity <= self.threshold_critical:
+                if (rospy.get_time() - self.last_time_warned) > self.interval_duration_critical:
+                    self.last_time_warned = rospy.get_time()
+                    mode = copy.copy(self.mode)
+                    mode.mode = LightModes.FLASH
+                    color = self.color_critical
+                    mode.colors = []
+                    mode.colors.append(color)
+                    mode.frequency = 4
+                    mode.pulses = 8
+                    self.set_light(mode)
+                    self.say(self.sound_critical)
+                if (rospy.get_time() - self.last_horn_time) > self.horn_interval_duration_critical:
```

Comment:
why don't you use the same interval duration here?

Reply:
The light warning interval is very short. It blinks every 5 seconds. If the robot would beep every five seconds too, people would go crazy. But if you want I can change that.